### ISSUE TITLE
Broken serialization when using Dict + serialization_strategy

### ISSUE THREAD

**Issue by zupo** [Fatal1ty/mashumaro#44]
I'm hitting a weird bug. When I use a `Dict` field type with a serialization_strategy, the `serialize` method is also passed the entire dict, not individual items of the dict. The result is broken serialization as every value contains all values. 

Confusing, right? I'll try to explain with an example.

```python
from enum import Enum
from typing import Set
from dataclasses import dataclass
from mashumaro import DataClassDictMixin
from typing import Dict
from decimal import Decimal
from mashumaro.config import BaseConfig
from mashumaro.types import RoundedDecimal


@dataclass()
class Foo(DataClassDictMixin):
    bar: Dict[str, Decimal]

    class Config(BaseConfig):
        serialization_strategy = {
            Decimal: RoundedDecimal(),
        }


foo = Foo(bar={"a": 1, "b": 2})
print(foo.to_dict())
assert foo.to_dict() == {"bar": {"a": "1", "b": "2"}}
```

This fails, because `foo.to_dict()` does not return `{"bar": {"a": "1", "b": "2"}}` but it returns `{'bar': {'a': "{'a': 1, 'b': 2}", 'b': "{'a': 1, 'b': 2}"}}`. See it there? Every value in the returned dict contains all values of the dict.

**Comment by Fatal1ty** [comment]
It looks like I made a mistake here https://github.com/Fatal1ty/mashumaro/blob/master/mashumaro/serializer/base/metaprogramming.py#L421

It should be `overridden = f"self.__{fname}_serialize({value_name})"`

**Comment by Fatal1ty** [comment]
Thanks for finding this bug! It was fixed in 2.0.2.

**Comment by zupo** [comment]
Wow, that was fast :o thanks!
